pkg/pool-utils/test/BasePool.test.ts

```diff
@@ -794,4 +794,54 @@ describe('BasePool', function () {
       }
     });
   });
+
+  describe('set asset manager config', () => {
+    let pool: Contract;
+    const poolConfig = {
+      targetPercentage: 3,
+      upperCriticalPercentage: 4,
+      lowerCriticalPercentage: 2,
+    };
+
+    const encodedConfig = ethers.utils.defaultAbiCoder.encode(
+      ['uint64', 'uint64', 'uint64'],
+      [bn(poolConfig.targetPercentage), bn(poolConfig.upperCriticalPercentage), bn(poolConfig.lowerCriticalPercentage)]
+    );
+
+    sharedBeforeEach('deploy pool and asset manager', async () => {
+      const assetManagerContract = await deploy('MockAssetManager', { args: [tokens.first.address] });
+
+      const assetManagers = Array(tokens.length).fill(ZERO_ADDRESS);
+      assetManagers[0] = assetManagerContract.address;
+
+      pool = await deployBasePool({ owner: poolOwner, assetManagers });
+    });
+
+    sharedBeforeEach('set permissions', async () => {
+      const pauseAction = await actionId(pool, 'pause');
+      await authorizer.connect(admin).grantPermissions([pauseAction], admin.address, [ANY_ADDRESS]);
+    });
+
+    it('lets the owner set the asset manager config', async () => {
+      await pool.connect(poolOwner).setAssetManagerPoolConfig(tokens.first.address, encodedConfig);
```

Comment:
Let's emit an event in the MockAssetManager with the pool id and data received, and check that they match the expected values.